x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportGetDataFrameTransformsStatsAction.java

```diff
@@ -212,6 +209,56 @@ private void collectStatsForTransformsWithoutTasks(Request request,
             }
         );
 
-        dataFrameTransformsConfigManager.getTransformStats(transformsWithoutTasks, searchStatsListener);
+        dataFrameTransformsConfigManager.getTransformStoredDoc(transformsWithoutTasks, searchStatsListener);
+    }
+
+    private void populateSingleStoppedTransformStat(DataFrameTransformStoredDoc transform,
+                                                    ActionListener<DataFrameTransformCheckpointingInfo> listener) {
+        transformsCheckpointService.getCheckpointStats(transform.getId(), transform.getTransformState().getCheckpoint(),
+            transform.getTransformState().getCheckpoint() + 1, transform.getTransformState().getIndexerState(),
+            transform.getTransformState().getPosition(), transform.getTransformState().getProgress(),
+            ActionListener.wrap(
+                listener::onResponse,
+                e -> {
+                    logger.warn("Failed to retrieve checkpointing info for transform [" + transform.getId() + "]", e);
+                    listener.onResponse(DataFrameTransformCheckpointingInfo.EMPTY);
+                }
+            ));
+    }
+
+    private void addCheckpointingInfoForTransformsWithoutTasks(List<DataFrameTransformStats> allStateAndStats,
+                                                               List<DataFrameTransformStoredDoc> statsForTransformsWithoutTasks,
+                                                               ActionListener<Void> listener) {
+
+        if (statsForTransformsWithoutTasks.isEmpty()) {
+            // No work to do, but we must respond to the listener
+            listener.onResponse(null);
+        }
+
+        AtomicInteger numberRemaining = new AtomicInteger(statsForTransformsWithoutTasks.size());
+        AtomicBoolean isExceptionReported = new AtomicBoolean(false);
+
+        statsForTransformsWithoutTasks.forEach(stat -> populateSingleStoppedTransformStat(stat,
```

Comment:
How does the cluster behave with a larger amount of transforms getting stats at the same time? This could be pretty bad if the UI is loading many transforms at the same time to display their stats and most of them are stopped.